openedx/core/djangoapps/programs/utils.py

```diff
@@ -341,34 +343,52 @@ def supplement_program_data(program_data, user):
             course_key = CourseKey.from_string(run_mode['course_key'])
             course_overview = CourseOverview.get_from_id(course_key)
 
-            run_mode['course_url'] = reverse('course_root', args=[course_key])
-            run_mode['course_image_url'] = course_overview.course_image_url
+            course_url = reverse('course_root', args=[course_key])
```

Comment:
I agree. The logic within this function is now getting complicated and can use helper methods.
Can you put a TODO comment on the top of this method?